lib/spack/spack/test/stage.py

```diff
@@ -536,11 +536,25 @@ def test_search_if_default_fails(self, failing_fetch_strategy, search_fn):
         with stage:
             try:
                 stage.fetch(mirror_only=False)
-            except spack.fetch_strategy.FetchError:
-                pass
+                assert False, 'Expected fetch to fail'
+            except spack.fetch_strategy.FetchError as err:
+                assert 'All fetchers failed' in str(err)
```

Comment:
I think it would be better to reformat this test like so:
```python
with pytest.raises(spack.fetch_strategy.FetchError) as err:
    stage.fetch(mirror_only=False)
assert 'All fetchers failed' in err.value
```
Same below. See https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions